src/Microsoft.Azure.WebJobs.Host/Indexers/FunctionIndexer.cs

```diff
@@ -311,13 +321,42 @@ private bool HasJobAttribute(CustomAttributeData attributeData)
             return new FunctionDefinition(descriptor, instanceFactory, listenerFactory);
         }
 
-        private static FunctionDescriptor CreateFunctionDescriptor(MethodInfo method, string triggerParameterName,
+        // Expose internally for testing purposes 
+        internal static FunctionDescriptor FromMethod(
```

Comment:
nit: `FromMethod` doesn't seem as descriptive to me, for example this usage is a bit unclear

`var descr = FromMethod(method, this._activator, _nameResolver);`